src/core/trtserver.h

```diff
@@ -758,6 +758,15 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerSharedMemoryAddress(
     TRTSERVER_Server* server, TRTSERVER_SharedMemoryBlock* shared_memory_block,
     size_t offset, size_t byte_size, void** base);
 
+/// Get the list of all active shared memory region on the inference server.
+/// If there are none then the list is empty. Returned error indicates if it
+/// was able to successfully get all active shared memory regions or not.
+/// \param server The inference server object.
+/// \param status Returns the shared memory status protobuf.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_ServerSharedMemoryStatus(
+    TRTSERVER_Server* server, TRTSERVER_Protobuf** status);
+
```

Comment:
"Get the current shared memory region status of the inference server. The caller takes ownership of 'status' and must call TRTSERVER_ProtobufDelete to release the object."